### ISSUE TITLE
If a node fails with dataflow error in batch run, it immediately falls back to a dry run afterwards

### ISSUE THREAD

**Issue by radeusgd** [enso-org/enso#7140]
See the video below:

https://github.com/enso-org/enso/assets/1436948/e9bfa027-737c-482e-91c7-5fe479055808

The Main file of the project used is as follows:
```
from Standard.Base import all
from Standard.Table import all
from Standard.Database import all
from Standard.AWS import all

import Standard.Base.Errors.Common.Dry_Run_Operation
import Standard.Base.Errors.Illegal_Argument.Illegal_Argument
import Standard.Base.Runtime.Context
import Standard.Visualization

my_operation name =
    case Context.Output.is_enabled of
        True -> if name == "will error" then Error.throw (Illegal_Argument.Error "The operation failed due to some reason.") else "Batch run succeeded"
        False ->
            result = "Dry run succeeded"
            warning = Dry_Run_Operation.Warning "But this is just a dry run"
            Warning.attach warning result

main =
    name1 = "foobar"
    op1 = my_operation name1
    text1 = "will error"
    operator1 = my_operation text1
```

We can see that first the nodes are in dry run mode - because we are live editing.

Once we hit the play button, both nodes do run. The first one runs and succeeds, returning a changed message. But the second one is more curious - it runs and results in an error (as planned). We can see that the error is reported as the node border turns red for a brief moment. But before we can even try to hover mouse over it, we can see that it is **re run again** back in dry-run mode and the error disappears, being replaced with the result of the dry run.

This makes it impossible to inspect any errors that occurred during a batch run. I also don't think this is what should happen in general - intuitively - I'd think that after the batch run the workflow should stay as it completed the batch run - any further updates of the workflow should be triggered by the user editing some nodes/inputs - **not automatically**.

**Comment by farmaazon** [comment]
We clearly does not meet a requirement from [the design](https://github.com/enso-org/design/blob/main/epics/basic-libraries/write-action-control/design.md#workflow-level-play-button):

> The flag on the engine overriding the mode permission is reset, but cached values are preserved. This means that the caches will contain the real data after the play button stops evaluating the graph.

Upon clicking "play" button, IDE sends "recompute" message (as expected), and after getting results attaches visualization for displaying error (also necessary part). Perhaps attaching the visualization causes re-computation.

Here I send a fragment of engine's verbose log, right after pressing play button:

[20230628-090032-219-enso-project-manager.log](https://github.com/enso-org/enso/files/11892472/20230628-090032-219-enso-project-manager.log)

**Comment by farmaazon** [comment]
So, I think we just should not recompute graph when attaching visualization (or compute the node only, if it's not cached somehow) - @4e6 or @JaroslavTulach do you agree?

**Comment by hubertp** [comment]
@farmaazon a side question - IDE triggers `recompute`, gets the expression updates, attaches visualization (1 rather than 2?!) and detaches it after a second. That sounds like a separate issue?

**Comment by farmaazon** [comment]
> attaches visualization (1 rather than 2?!) and detaches it after a second. That sounds like a separate issue?

This is expected behavior.

The attached visualization is for displaying Dataflow Errors. So we get information from the engine that a node has dataflow error and we attach visualization to get the error message. But in the second run the error is no more, so we detach visualization.
